### ISSUE TITLE
Pasting same-page link adds admin page URL

### ISSUE THREAD

**Issue by sabernhardt** [WordPress/gutenberg#28115]
**Original report by hage**
[Trac #52105](https://core.trac.wordpress.org/ticket/52105)

**Describe the bug**
When copying and pasting anchor links, the links have an incorrect destination by adding the page's admin URL.

**Steps to reproduce the behavior:**
1. Create text within a rich text block (paragraph, heading, or list).
2. Create a link on all or part of that text, so that it leads to a section on the same page (`#destination`).
3. Either cut or copy the link.
4. Paste the link in another block or another part of the same block.
5. Click on the link to open the link editor dialog and see the full URL of the admin page for the new link's target location plus `#destination`.

**Expected behavior**
The link should remain a same-page link.

**Screenshots**

Before moving:
![link is only the target of `#entwicklung`](https://core.trac.wordpress.org/raw-attachment/ticket/52105/bug_anchor_3.jpg)

After moving:
![link includes admin URL](https://core.trac.wordpress.org/raw-attachment/ticket/52105/bug_anchor_2.jpg)

Another screenshot after pasting (WP 5.6, without Gutenberg or Yoast plugins):
![WordPress 5.6, after pasting](https://user-images.githubusercontent.com/17100257/104233241-282dc000-5417-11eb-95fd-562476e1a77f.jpg)

**WordPress versions**
- 5.6 without Gutenberg plugin
- also 5.0 without Gutenberg
- also 5.7 trunk (49950) with Gutenberg 9.7

**Desktop (please complete the following information):**
 - OS: Windows 10, 1909
 - Browsers: Chrome 87.0.4280.88 and Firefox 84.0.2

**Comment by aristath** [comment]
@sabernhardt I couldn't replicate this issue... Does this also happen when manually entering the URL instead of copy-pasting it? Where exactly was the anchor URL copied from? I tried copying it from the "HTML Anchor" field of another block but there was no issue when I tried that either. 🤔

**Comment by sabernhardt** [comment]
Sorry for the confusion. I did not mean the HTML Anchor field.

Hopefully this clarifies it a bit:
1. When creating a text link, manually type the destination to direct to an anchor elsewhere in the same post.
2. Then cut or copy that text link, and paste it in another place within the post content.
3. The link URL no longer matches what you typed.

**Comment by paaljoachim** [comment]
Testing using WordPress 5.7
Gutenberg plugin 10.2.1.
Twenty Twenty One.

Added an internal link. Copied the text that had the link. Pasted the text and link into another text block.
I am seeing an identical link.

**Comment by gwwar** [comment]
@sabernhardt Do you mean this flow?

https://user-images.githubusercontent.com/1270189/112534621-81e02e00-8d68-11eb-8004-e939a0cf0318.mp4

For other copy/duplication paths it looks okay:

https://user-images.githubusercontent.com/1270189/112534667-93293a80-8d68-11eb-8503-cda3a3285825.mp4

**Comment by paaljoachim** [comment]
This issue might be associated: 
Insert Link: clicking on same-page links reloads the editor
https://github.com/WordPress/gutenberg/issues/10117

**Comment by sabernhardt** [comment]
It seems this has been fixed elsewhere.

I've used the Ctrl + C/V keyboard shortcuts for copying and pasting the linked text, so it's basically the same as the browser context menu (flow 1).

I tried again today with 5.7 in Firefox 86.0.1, both with and without Gutenberg 10.2.1 activated.
- With the plugin, it pastes the link without the full URL.
- Without the plugin, it still pastes the full admin URL with the anchor.

**Comment by skorasaurus** [comment]
Still able to reproduce this in Gutenberg 10.9.0.

**Comment by matthiaspabst** [comment]
I can't reproduce this in WordPress 5.9+. Seems like this is fixed now.

**Comment by skorasaurus** [comment]
Neither am I able to reproduce; closing.

**Comment by matthiaspabst** [comment]
The bug still exists (or exists again?) in WordPress 6.0.2.

**Comment by smerriman** [comment]
Just came up against this today myself. Can this be reopened or should I open a new issue?

**Comment by skorasaurus** [comment]
> Just came up against this today myself. Can this be reopened or should I open a new issue?

I reopened this but could you share more information about your environment; do you have the Gutenberg plugin installed? Which version of WordPress are you using? I am unable to reproduce this myself in WP 6.2 with Gutenberg 15.6.2 installed. 

Video shows that the copied link is reproduced correctly 

https://user-images.githubusercontent.com/955351/236244704-f8ae19d0-c4d7-45dd-9862-f0a93b40d820.mp4

**Comment by smerriman** [comment]
Apologies for the delay - it happened for me 3 times while I was migrating a site the other week, but I can't recall the exact circumstances and now am unable to replicate; I assumed based on matthiaspabst's message that it was happening more generically. Will follow up the next time it happens.

**Comment by smerriman** [comment]
I've managed to replicate the issue. It now only exists with a cut and paste, rather than a copy and paste.

https://watch.screencastify.com/v/7Bf2BHOZJ9pufLCF1I3K

- Enter some text, link it to anchor
- Select it, press Ctrl X.
- If you press Ctrl-V immediately to re-paste it into the same paragraph, it works
- However, it you select any other paragraph and paste it, it breaks with the admin URL being appended.

So basically, cut and paste always fails, except if re-pasting immediately where you cut it from.

The link popup appears very buggy while cutting and pasting as well, appearing partially offscreen even though there is no link visible.

edit - for reference, this is WP 6.2.1, no Gutenberg plugin.
